lib/auth/join_test.go

```diff
@@ -250,6 +253,16 @@ func TestAuth_RegisterUsingToken(t *testing.T) {
 			certs, err := a.RegisterUsingToken(ctx, tc.req)
 			if tc.errorAssertion != nil {
 				require.True(t, tc.errorAssertion(err))
+				if tc.errorAssertion != nil {
+					require.True(t, tc.errorAssertion(err))
+					if tc.waitTokenDeleted {
+						require.Eventually(t, func() bool {
+							_, err := a.ValidateToken(ctx, tc.req.Token)
+							return err != nil && strings.Contains(err.Error(), "not found")
```

Comment:
I think you can use `trace.IsNotFound` instead of comparing strings.

Reply:
It's actually access denied, I'll use const from auth for checking text.